### PR TITLE
feat: dir and file flags that validate existence and type

### PR DESCRIPTION
**Pull request by mshanemc**
dir and file flags from https://github.com/salesforcecli/sf-plugins-core/pull/32

original WI: [@W-10766801@](https://gus.lightning.force.com/a07EE00000roHVgYAM)

### CODE REVIEW

File: src/parser/flags.ts [hunk 1]
```diff
@@ -42,6 +43,20 @@ export const integer = build({
   },
 })
 
+/**
+ * Accepts a directory path.  Validates that the directory exists and is a directory.
+ */
+export const existingDirectory = build<string>({
```
Comment (mdonnalley) [inline src/parser/flags.ts:49]: What do you think about changing these to `directory` and `file` instead? And then you can specify `exists: true` in the configuration if you need the directory or file to exist

**Review by mdonnalley — CHANGES_REQUESTED** [review]

File: src/parser/flags.ts (hunk 1 above)
Comment (mshanemc) [reply src/parser/flags.ts]: what would they do if they weren't `exists: true` ?

Also, none of the other other flags have a configuration beyond the OptionFlag stuff.  We *could* make them more like what we're doing in the sf-core-plugins where there's a need to pass in extra parameters (wrapping flag.build and returning the result of calling it) but that's not how the other oclif flags formed from OptionFlag work.

Comment (RodEsp) [reply src/parser/flags.ts]: If `exists: false` it would just validate that you're giving it a valid dir/file path.

**Review by mdonnalley — APPROVED** [review]

File: src/parser/flags.ts [hunk 2]
```diff
@@ -42,6 +43,20 @@ export const integer = build({
   },
 })
 
+export const directory = (opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> => {
+  return build<string>({
+    ...opts,
+    parse: async (input: string) => opts.exists ? dirExists(input) : input,
```
Comment (RodEsp) [inline src/parser/flags.ts:49]: If `exists` is `false` this should at least validate the input is a valid directory path.

File: src/parser/flags.ts [hunk 3]
```diff
@@ -42,6 +43,20 @@ export const integer = build({
...
+  })()
+}
+
+export const file = (opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> => {
+  return build<string>({
+    ...opts,
+    parse: async (input: string) => opts.exists ? fileExists(input) : input,
```
Comment (RodEsp) [inline src/parser/flags.ts:56]: If `exists` is `false` this should at least validate the input is a valid file path.

**Review by RodEsp — CHANGES_REQUESTED** [review]

File: src/parser/flags.ts (hunk 2 above)
Comment (mshanemc) [reply src/parser/flags.ts:49]: can you explain what these validation would do (ex: what's valid vs. invalid?). Like certain characters that can't be used depending on the os?

Comment (RodEsp) [reply src/parser/flags.ts:49]: @mshanemc yep, exactly that.

**Review by RodEsp — APPROVED** [review]

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**